sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1449,6 +1451,16 @@ object SQLConf {
     .intConf
     .checkValues((1 to 9).toSet + Deflater.DEFAULT_COMPRESSION)
     .createWithDefault(Deflater.DEFAULT_COMPRESSION)
+
+  val AVRO_XZ_LEVEL = buildConf("spark.sql.avro.xz.level")
```

Comment:
Just for clarification, @MaxGekk, this configuration is not for keeping the configuration in spark-avro (as a third party) but something you here newly propose?

Reply:
Yes, it is a new feature. The third party doesn't support `LZMA2` codec. The last one has much better compression than `snappy`/`deflate` even on level `0-3`. The default level `6` covers most of use cases actually when users need to store data in AVRO format permanently as an archive. If you strongly against of the config, I will remove it.

Reply:
> If you faced a case that this configuration is useful, im okay but if not, I would suggest to add it later when it's needed.

The config is not so useful as `LZMA2` codec itself. Its default value covers the use case. I will remove the config from the PR.